cmd/timestamp_check/main.go

```diff
@@ -0,0 +1,164 @@
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/config"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/signaler"
+)
+
+func main() {
+	flag.BoolVar(&verbose, "verbose", false, "verbose output")
+	flag.Parse()
+
+	log.Println("Checking GCT supported exchanges for timestamp issues on authenticated endpoints.")
+	log.Println()
+
+	if !LoadConfig() {
+		os.Exit(0)
+	}
+
+	var wg sync.WaitGroup
+	for i := range SupportedExchanges {
+		SupportedExchanges[i].SetDefaults()
+		cfg, ok := configs[SupportedExchanges[i].GetName()]
+		if !ok {
+			cfg.Report = &Report{errors.New("cannot find configuration for exchange")}
+			continue
+		}
+
+		if !cfg.Enabled {
+			cfg.Report = &Report{fmt.Errorf("not enabled in %s", configFilename)}
+			continue
+		}
+
+		if cfg.Keys == (Keys{}) {
+			cfg.Report = &Report{fmt.Errorf("API Keys not set in %s", configFilename)}
+			continue
+		}
+
+		err := SupportedExchanges[i].Setup(&config.ExchangeConfig{
+			Enabled: true,
+			Verbose: verbose,
+			API: config.APIConfig{
+				AuthenticatedSupport: true,
+				Credentials: config.APICredentialsConfig{
+					Key:       cfg.Key,
+					Secret:    cfg.Secret,
+					ClientID:  cfg.ClientID,
+					PEMKey:    cfg.PEMKey,
+					OTPSecret: cfg.OTP,
+				},
+				CredentialsValidator: &config.APICredentialsValidatorConfig{},
+			},
+		})
+		if err != nil {
+			cfg.Report = &Report{err}
+			continue
+		}
+
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, exch exchange.IBotExchange, cfg *TimestampConfiguration) {
+			defer wg.Done()
+			_, err := exch.FetchAccountInfo()
+			if err != nil {
+				if err != nil {
+					cfg.Report = &Report{err}
+					return
+				}
+			}
+
+			log.Printf("%s initial nonce/timestamp created, sleeping for %s\n",
+				exch.GetName(),
+				defaultSleepTime)
+
+			time.Sleep(defaultSleepTime)
+			_, err = exch.FetchAccountInfo()
+			if err != nil {
+				if err != nil {
+					cfg.Report = &Report{err}
+					return
+				}
+			}
+			cfg.Report = &Report{}
+		}(&wg, SupportedExchanges[i], cfg)
+	}
+
+	interrupted := make(chan struct{})
+	go func(ch chan struct{}) {
+		signaler.WaitForInterrupt()
+		ch <- struct{}{}
+	}(interrupted)
+
+	finished := make(chan struct{})
+	go func(ch chan struct{}, wg *sync.WaitGroup) {
+		wg.Wait()
+		ch <- struct{}{}
+	}(finished, &wg)
+
+	select {
+	case <-interrupted:
+		log.Println("Interruption caught, shutting down...")
+	case <-finished:
+	}
+
+	log.Println()
+	log.Println("Report:")
+	for key, val := range configs {
+		if val.Report != nil {
+			if val.Report.Error != nil {
+				log.Printf("%s has [FAILED] authentication validation: %s",
+					key,
+					val.Report.Error)
+				continue
+			}
+			log.Printf("%s has [PASSED] authentication validation", key)
+			continue
+		}
+		log.Printf("%s was not able to test authentication validation", key)
+	}
+}
+
+// LoadConfig loads configuration
+func LoadConfig() bool {
+	fileData, err := ioutil.ReadFile(configFilename)
```

Comment:
Should use file.Exists instead (or os.Stat combined and checking err == IsNotExist) of this so you can specifically check to see if it's missing